infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java

```diff
@@ -141,23 +141,20 @@ void prepare(boolean markManaged, boolean canCreate) throws InfrastructureExcept
 
   /**
    * Deletes the namespace. Deleting a non-existent namespace is not an error as is not an attempt
-   * to delete a namespace that is already being deleted.
+   * to delete a namespace that is already being deleted. If the namespace is not marked as managed,
+   * it is silently not deleted.
    *
-   * @throws InfrastructureException if the namespace is not marked managed or when any unexpected
-   *     exception occurs during namespace deletion
+   * @throws InfrastructureException f any unexpected exception occurs during namespace deletion
    */
   void deleteIfManaged() throws InfrastructureException {
     KubernetesClient client = clientFactory.create(workspaceId);
 
-    if (!isManagedInternal(client)) {
-      throw new InfrastructureException(
-          format(
-              "Can't delete namespace '%s' that contains"
-                  + " runtime of workspace '%s' because it doesn't have the '"
-                  + MANAGED_NAMESPACE_LABEL
-                  + "' label equal to 'true'.",
-              name,
-              workspaceId));
+    if (!isNamespaceManaged(client)) {
+      LOG.debug(
+          "Namespace {} for workspace {} is not marked as managed. Ignoring the delete request.",
+          name,
+          workspaceId);
+      return;
     }
 
     try {
```

Comment:
can we catch delete exception with 403 check in delete method?

Reply:
Let's solve this under https://github.com/eclipse/che/issues/15250